R/plot_sampling_hist.R

```diff
@@ -17,5 +17,53 @@
 #' samples <- draw_samples(pop, 3, c(1, 10))
 #' plot_sampling_hist(pop, samples, var_name, c(1, 10), 3)
 plot_sampling_hist <- function(pop, samples, var_name, n_s, reps){
+  if (!"data.frame" %in% c(class(samples))) stop("Samples should be a data frame or a tibble")
+  if (!is.numeric(reps) == TRUE) stop("Number of replications should be a numerical value, a vector with length 1")
+  for (i in n_s){
+    if (class(i) != "numeric") stop("Samples' sizes should be a list or a vector with only numeric values")
+  }
+
+  samples <-
+    samples %>%
+    ungroup()
+
+  x_min <-
+    samples %>%
+    select({{var_name}}) %>%
+    unlist() %>%
+    quantile(0.05)
+
+  x_max <-
+    samples %>%
+    ungroup() %>%
+    select({{var_name}}) %>%
+    unlist() %>%
+    quantile(0.95)
+
+
+  summary <-
+    samples %>%
+    group_by(replicate, size, rep_size) %>%
+    summarise(mean = mean({{var_name}}))
+
+
+  sampling_dist<- list()
+  for (i in 1:length(n_s)){
+    sampling_dist[[ i ]] <-
+      summary %>%
+      filter(size == n_s[i]) %>%
+      ggplot() + geom_histogram(aes(mean, ..density..)) +
```

Comment:
I think you need to add ggplot to the dependence list
refer: https://r-pkgs.org/whole-game.html#use_package

Reply:
Thanks! I'll take a look at it.

Reply:
Looks good